packages/next/src/build/webpack-config.ts

```diff
@@ -97,6 +97,10 @@ const reactPackagesRegex = /^(react|react-dom|react-server-dom-webpack)($|\/)/
 const asyncStoragesRegex =
   /next[\\/]dist[\\/]client[\\/]components[\\/](static-generation-async-storage|action-async-storage|request-async-storage)/
 
+// exports.<conditionName>
+const edgeConditionNames = ['edge-light', 'worker', 'browser', 'module']
```

Comment:
@sokra seems missing `"import"` here breaks the resolving of the modules. Does this change the behavior of resolving with import type by default?

Reply:
`import` is different because it's not a condition of the environment (am I running in a browser? edge? regular node?) but of the way the code is written (do I use `import` syntax or `require()` calls?). If removing `import` leads to unexpected results, then your code is using `require()` calls.

You shouldn't override this, because the imported code could be written in a way designed for CJS or ESM semantics. Eg, I usually `module.exports = something` for CJS, but use `export default something` for ESM. These have different meanings at the importing site, and so the user's code can break if the correct one isn't returned. (Getting the ESM code when using `require()` will mean the user has to `require().default`, and getting CJS code when using `import` syntax will mean the user can't use named imports).

Reply:
👍 Great explanation! Turns out we're using require in edge loader which breaks the behavior, thanks